moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -569,31 +578,22 @@ class KinematicsBase
   /**
    * @brief  Virtual destructor for the interface
    */
-  virtual ~KinematicsBase()
-  {
-  }
+  virtual ~KinematicsBase();
 
-  KinematicsBase()
-    : tip_frame_("DEPRECATED")
-    ,  // help users understand why this variable might not be set
-       // (if multiple tip frames provided, this variable will be unset)
-    search_discretization_(DEFAULT_SEARCH_DISCRETIZATION)
-    , default_timeout_(DEFAULT_TIMEOUT)
-  {
-    supported_methods_.push_back(DiscretizationMethods::NO_DISCRETIZATION);
-  }
+  KinematicsBase();
 
 protected:
+  moveit::core::RobotModelConstPtr robot_model_;
   std::string robot_description_;
   std::string group_name_;
   std::string base_frame_;
   std::vector<std::string> tip_frames_;
-  std::string tip_frame_;  // DEPRECATED - this variable only still exists for backwards compatibility with
-                           // previously generated custom ik solvers like IKFast
 
-  double search_discretization_;  // DEPRECATED - this variable only still exists for backwards compatibility
-                                  // with previous implementations.  Discretization values for each joint are
-                                  // now stored in the redundant_joint_discretization_ member
+  // The next two variables still exists for backwards compatibility
+  // with previously generated custom ik solvers like IKFast
+  // Replace tip_frame_ -> tip_frames_[0], search_discretization_ -> redundant_joint_discretization_
+  MOVEIT_DEPRECATED std::string tip_frame_;
+  MOVEIT_DEPRECATED double search_discretization_;
```

Comment:
i didn't know you could MOVEIT_DEPRECATED a variable. how does this change the behavior of anything?

Reply:
You should not use the variable anymore, otherwise you get a deprecation warning.